lib/datasource.js

```diff
@@ -456,11 +456,21 @@ DataSource.prototype.defineRelations = function(modelClass, relations) {
     }
   };
 
+  var hasInvalidRelationName = function(relationName) {
```

Comment:
Since this function does not need anything from `defineRealations` closure, it's better to move it to the root scope of this file.

Also `isValidRelationName` is a better name, as it prevents double-negative in `!hasInvalidRelationName`.